cluster-state-service/internal/features/e2einstancesteps/list_instances_steps.go

```diff
@@ -73,10 +73,32 @@ func init() {
 	})
 
 	Then(`^the list instances response contains all the instances registered with the cluster$`, func() {
-		if len(ecsContainerInstanceList) != len(cssContainerInstanceList) {
+		cssContainerInstanceARNs := make([]*string, 0, len(cssContainerInstanceList))
+		for _, v := range cssContainerInstanceList {
```

Comment:
If we have to go through the loop anyway, lets just do it once below and compare both arn and the status. We dont have to do the length check.